src/test/ui/hygiene/hygienic-labels-in-let.rs

```diff
@@ -63,7 +53,6 @@ pub fn main() {
     let k: isize = {
         'x: for _ in 0..1 {
```

Comment:
@petrochenkov I mean this still warns because of the `'x: loop {` above.

Reply:
Oh, wait, due to this thing?
```rust
                if rib.kind.is_label_barrier() {
                    rib.bindings.insert(ident, id);
                    break;
                }
```
I.e. looks like we are basically define the label at the whole function level, so it still produces the warning in the unrelated cases. We certainly shouldn't do that.

I no longer understand what this PR removes then.
That's why I asked to implement just the removal without the AST move in https://github.com/rust-lang/rust/pull/96296#issuecomment-1106559454.

Reply:
I expected the behavior described in https://github.com/rust-lang/rust/pull/96296#issuecomment-1106576438 and https://github.com/rust-lang/rust/pull/96296#issuecomment-1114180687.
I.e. only shadowing, and only in the same namespace is still diagnosed.

Reply:
Corrected our misunderstanding. Now, the PR does what advertised.